system-contract-dapp-playground/src/components/contract-interaction/erc/utils/methodInterfaces.ts

```diff
@@ -0,0 +1,39 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { ethers } from 'ethers';
+
+/**
+ * @notice interface for methods from ERC20Mock contract
+ */
+export interface ERC20MockMethod extends ethers.BaseContract {
```

Comment:
Q: Is there a way to utilize the ABI instead of having to define a new interface?

Reply:
Not that I know of. So in the last PR, the `generateBaseContractInstance()` generates a new instance of BaseContract like this 
```
const baseContract = new ethers.BaseContract(
      contractAddress,
      JSON.stringify(contractABI),
      walletSigner
    );
```
and without the `ERC20MockMethod` interface above, typescript would not recognize the methods from the smart contracts that's why I created this `ERC20MockMethod` interface

Reply:
🤔 okay.
The challenge is by introducing a new interface you now introduce something that must be maintained. It's simple in this case since ERC20 won't expand but I imagine this strategy is going to be used going forward for HTS.
So now this interface would have to be updated in the HTS case.
An option that utilizes the ABI would be nice but if there's none that's understandable.
Do some exploration to see if there's a way to go from ABI to interface in `ts`, if not as you have it is fine

Reply:
Gotcha!